Add new rule to Mulan PubL v1 License

Tang-Kin-Long opened the pull request:
Hi, I am the contributor from RecLicense project of osslab-pku, taking on the issue "[Contribute updates back to ScanCode?](https://github.com/osslab-pku/RecLicense/issues/2)".
I observe that in Mulan License family, **the rule for detecting Mulan Public License, Version 1 (Mulan PubL v1) is not existed**.
I've done some tests in local environment to show that existing rules for Mulan PSL v1, ,Mulan PSL v2, Mulan PubL v2 can adequately detect and distinguish Mulan license.
Therefore, currently I only consider to add a new rule [mulanpubl-1.0_1.RULE](https://github.com/aboutcode-org/scancode-toolkit/pull/4627/commits/b6444ee6f30b546dbb9edf2e6cefd05c7c5f3949) for Mulan PubL v1, which takes the content of [mulanpubl-2.0_1.RULE](https://github.com/aboutcode-org/scancode-toolkit/blob/develop/src/licensedcode/data/rules/mulanpubl-2.0_1.RULE) as reference.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [ ] Updated documentation pages (if applicable)
* [ ] Updated CHANGELOG.rst (if applicable)

Signed-off-by: Jianlang Deng <csgarytang@gmail.com>

Tang-Kin-Long committed:
Add new rule to Mulan PubL v1 License

Signed-off-by: Jianlang Deng <csgarytang@gmail.com>

minghuizhou commented:
Still an open issue. @alex @aslakhellesoy

pombredanne commented:
@Tang-Kin-Long Thanks! If you see more examples of these notices that would be great!

AyanSinhaMahapatra commented on src/licensedcode/data/rules/mulanpubl-1.0_1.RULE:
```diff
@@ -0,0 +1,15 @@
+    - http://license.coscl.org.cn/MulanPubL-1.0
+---
+
+licensed under Mulan PubL v1.
```
```suggestion
licensed under {{Mulan PubL v1}}.
```

Tang-Kin-Long replied:
Thank you for your suggestion. I'm very sorry that I ignored the DCO inspection requirements in this commit due to my hurry, and I will fix it later. But there are another 4 failing checks during submission. May I ask if these 4 failing checks will impact this commit?

AyanSinhaMahapatra replied:
These are unrelated and are being fixed by another PR, don't worry about them. :)

AyanSinhaMahapatra commented on src/licensedcode/data/rules/mulanpubl-1.0_1.RULE:
```diff
@@ -0,0 +1,15 @@
+licensed under Mulan PubL v1.
+You can use this software according to the terms and conditions of the Mulan PubL v1.
+You may obtain a copy of Mulan PubL v1 at:
+         http://license.coscl.org.cn/MulanPubL-1.0
```
```suggestion
         {{http://license.coscl.org.cn/MulanPubL-1.0}}
```

AyanSinhaMahapatra reviewed (approved):
Thanks++ @Tang-Kin-Long Looks good.

Could you also add these required phrases on the rule text so we don't detect similar texts on a partial match from this license rule?

I would also make sense to add another license reference rule with `http://license.coscl.org.cn/MulanPubL-1.0` as the rule text.

Tang-Kin-Long committed:
Update src/licensedcode/data/rules/mulanpubl-1.0_1.RULE

add required phrases "{{Mulan PubL v1}}" on rule text

Co-authored-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

Tang-Kin-Long committed:
Update src/licensedcode/data/rules/mulanpubl-1.0_1.RULE

add required phrases "{{http://license.coscl.org.cn/MulanPubL-1.0}}" on rule text
Signed-off-by: Jianlang Deng <csgarytang@gmail.com>

Co-authored-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>
